src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -22769,6 +22769,30 @@ Compiler::fgWalkResult Compiler::fgUpdateInlineReturnExpressionPlaceHolder(GenTr
         }
 #endif // DEBUG
 
+        var_types newType = inlineCandidate->TypeGet();
+
+        // If we end up swapping type we may need to retype the tree:
```

Comment:
Prior to this comment, neither the function header nor the code above talk about swapping the type. It would be good to describe why, and in what scenarios, such type swapping is done (including why `retType` may not be the same as `newType`).